b2sdk/utils/thread_pool.py

```diff
@@ -0,0 +1,45 @@
+from concurrent.futures import ThreadPoolExecutor
+from threading import Lock
+from typing import TYPE_CHECKING
+
+from b2sdk.utils import B2TraceMetaAbstract
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+class LazyThreadPool(metaclass=B2TraceMetaAbstract):
+    """
+    TODO: Add docstring
+    """
+
+    def __init__(self, max_workers: 'Optional[int]' = None) -> None:
+        """
+        :param Optional[int] max_workers: maximum number of worker threads
+        """
+
+        self._lock = Lock()
+        self._thread_pool: 'Optional[ThreadPoolExecutor]' = None
+        self._max_workers = max_workers
+
+    def set_thread_pool_size(self, max_workers: int) -> None:
```

Comment:
is this used anywhere? I think it's not. But if it is, then I think get_thread_pool should also use the same lock

Reply:
agreed, `get` needs to use the same lock